File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/webapp/src/ds/FinancialAccountMethodDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.md.org.ds.model.FinancialAccountMethodDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"financialAccountId", type:"int", useNull:true},
		{name:"financialAccount", type:"string"},
		{name:"financialAccountType", type:"string"},
		{name:"payMethodId", type:"int", useNull:true},
		{name:"payMethod", type:"string"},
		{name:"allowPayIn", type:"boolean"},
		{name:"allowPayOut", type:"boolean"}
	]
});

Ext.define("net.nan21.dnet.module.md.org.ds.model.FinancialAccountMethodDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"financialAccountId", type:"int", useNull:true},
		{name:"financialAccountId_From",type:"int", useNull:true},
		{name:"financialAccountId_To",type:"int", useNull:true},
		{name:"financialAccount", type:"string"},
		{name:"financialAccountType", type:"string"},
		{name:"payMethodId", type:"int", useNull:true},
		{name:"payMethodId_From",type:"int", useNull:true},
		{name:"payMethodId_To",type:"int", useNull:true},
		{name:"payMethod", type:"string"},
		{name:"allowPayIn", type:"boolean", useNull:true},
		{name:"allowPayOut", type:"boolean", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.md.org.ds.param.FinancialAccountMethodDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/org/ds/model/FinancialAccountMethodDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.md.org.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;
import net.nan21.dnet.module.md.org.domain.entity.FinancialAccountMethod;

@Ds(entity = FinancialAccountMethod.class)
public class FinancialAccountMethodDs extends
        AbstractAuditableDs<FinancialAccountMethod> {

    public static final String fFINANCIALACCOUNTID = "financialAccountId";
    public static final String fFINANCIALACCOUNT = "financialAccount";
    public static final String fFINANCIALACCOUNTTYPE = "financialAccountType";
    public static final String fPAYMETHODID = "payMethodId";
    public static final String fPAYMETHOD = "payMethod";
    public static final String fALLOWPAYIN = "allowPayIn";
    public static final String fALLOWPAYOUT = "allowPayOut";

    @DsField(join = "left", path = "financialAccount.id")
    private Long financialAccountId;

    @DsField(join = "left", path = "financialAccount.name")
    private String financialAccount;

    @DsField(join = "left", path = "financialAccount.type")
    private String financialAccountType;

    @DsField(join = "left", path = "payMethod.id")
    private Long payMethodId;

    @DsField(join = "left", path = "payMethod.name")
    private String payMethod;

    @DsField()
    private Boolean allowPayIn;

    @DsField()
    private Boolean allowPayOut;

    public FinancialAccountMethodDs() {
        super();
    }

    public FinancialAccountMethodDs(FinancialAccountMethod e) {
        super(e);
    }

    public Long getFinancialAccountId() {
        return this.financialAccountId;
    }

    public void setFinancialAccountId(Long financialAccountId) {
        this.financialAccountId = financialAccountId;
    }

    public String getFinancialAccount() {
        return this.financialAccount;
    }

    public void setFinancialAccount(String financialAccount) {
        this.financialAccount = financialAccount;
    }

    public String getFinancialAccountType() {
        return this.financialAccountType;
    }

    public void setFinancialAccountType(String financialAccountType) {
        this.financialAccountType = financialAccountType;
    }

    public Long getPayMethodId() {
        return this.payMethodId;
    }

    public void setPayMethodId(Long payMethodId) {
        this.payMethodId = payMethodId;
    }

    public String getPayMethod() {
        return this.payMethod;
    }

    public void setPayMethod(String payMethod) {
        this.payMethod = payMethod;
    }

    public Boolean getAllowPayIn() {
        return this.allowPayIn;
    }

    public void setAllowPayIn(Boolean allowPayIn) {
        this.allowPayIn = allowPayIn;
    }

    public Boolean getAllowPayOut() {
        return this.allowPayOut;
    }

    public void setAllowPayOut(Boolean allowPayOut) {
        this.allowPayOut = allowPayOut;
    }

}
